[BUG] npx 'canceled' error reason lacks clarity

### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

Using npm version 9.5.0 running the command:

```bash
npx --no-install esbuild --bundle "~/typescript-graphql-service/service/src/lambda.ts" --target=node18 --platform=node --outfile="~/typescript-graphql-service/AWS/cdk.out/bundling-temp-d2395ea5f5b16624b9255daf70671b29ee231644c99ea4f7718828f55c1a0f13/index.js" --external="@aws-sdk/*" --loader:.graphql=file
```

I get 

```bash
npm ERR! canceled

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lancegliser/.npm/_logs/2023-03-01T18_11_17_043Z-debug-0.log
```

### Expected Behavior

A more useful explanation of why it was cancelled. It was unclear what module was throwing an error or canceling at first.

```bash
npm ERR! npx canceled due to missing packages and no YES option: ["esbuild@0.17.10"]

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lancegliser/.npm/_logs/2023-03-01T18_11_17_043Z-debug-0.log
```

### Steps To Reproduce

1. In a general termal
2. Without `esbuild` in path
3. Run `npx --no-install esbuild --bundle`
4. See error...


### Environment

- npm: 9.5.1
- Node.js: 18.13.0
- OS Name: MacOS 12.6.3
- System Model Name: Mackbook Pro
- npm config: 
```ini
; "user" config from /Users/lancegliser/.npmrc

init-author-name = "Lance Gliser" 

; node bin location = /usr/local/bin/node
; node version = v18.13.0
; npm local prefix = /Users/lancegliser
; npm version = 9.5.1
; cwd = /Users/lancegliser
; HOME = /Users/lancegliser
; Run `npm config ls -l` to show all defaults.```

--- a/test/lib/commands/exec.js
+++ b/test/lib/commands/exec.js
@@ -3,8 +3,6 @@ const fs = require('fs/promises')
 const path = require('path')
 const { load: loadMockNpm } = require('../../fixtures/mock-npm.js')
 const MockRegistry = require('@npmcli/mock-registry')
-const { packument } = require('pacote')
-const { version } = require('os')
 
 t.test('call with args', async t => {
   const { npm } = await loadMockNpm(t, {

--- a/test/lib/commands/exec.js
+++ b/test/lib/commands/exec.js
@@ -136,7 +136,7 @@ t.test('npx --no-install @npmcli/npx-test', async t => {
     registry: 'https://registry.npmjs.org/',
   })
 
-  const manifest = registry.manifest({ name: '@npmcli/npx-test'})
+  const manifest = registry.manifest({ name: '@npmcli/npx-test' })
   manifest.versions['1.0.0'].bin = { 'npx-test': 'index.js' }
 
   const { npm } = await loadMockNpm(t, {
@@ -159,7 +159,7 @@ t.test('npx --no-install @npmcli/npx-test', async t => {
   } catch (error) {
     t.match(
       error.message,
-      'npx canceled due to missing packages and no YES option: ', 
+      'npx canceled due to missing packages and no YES option: ',
       'Expected error message thrown'
     )
   }

--- a/test/lib/commands/exec.js
+++ b/test/lib/commands/exec.js
@@ -3,6 +3,8 @@ const fs = require('fs/promises')
 const path = require('path')
 const { load: loadMockNpm } = require('../../fixtures/mock-npm.js')
 const MockRegistry = require('@npmcli/mock-registry')
+const { packument } = require('pacote')
+const { version } = require('os')
 
 t.test('call with args', async t => {
   const { npm } = await loadMockNpm(t, {
@@ -128,4 +130,39 @@ t.test('workspaces', async t => {
   await npm.exec('exec', ['@npmcli/npx-test'])
   const exists = await fs.stat(path.join(npm.prefix, 'workspace-a', 'npm-exec-test-success'))
   t.ok(exists.isFile(), 'bin ran, creating file inside workspace')
+})
+
+t.test('npx --no-install @npmcli/npx-test', async t => {
+  const registry = new MockRegistry({
+    tap: t,
+    registry: 'https://registry.npmjs.org/',
+  })
+
+  const manifest = registry.manifest({ name: '@npmcli/npx-test'})
+  manifest.versions['1.0.0'].bin = { 'npx-test': 'index.js' }
+
+  const { npm } = await loadMockNpm(t, {
+    config: {
+      audit: false,
+      yes: false,
+    },
+    prefixDir: {
+      'npm-exec-test': {
+        'package.json': JSON.stringify(manifest),
+        'index.js': `#!/usr/bin/env node
+  require('fs').writeFileSync('npm-exec-test-success', '')`,
+      },
+    },
+  })
+
+  try {
+    await npm.exec('exec', ['@npmcli/npx-test'])
+    t.fail('Expected error was not thrown')
+  } catch (error) {
+    t.match(
+      error.message,
+      'npx canceled due to missing packages and no YES option: ', 
+      'Expected error message thrown'
+    )
+  }
 })

--- a/workspaces/libnpmexec/lib/index.js
+++ b/workspaces/libnpmexec/lib/index.js
@@ -249,6 +249,7 @@ const exec = async (opts) => {
         // set -n to always say no
         if (yes === false) {
           // Error message lists missing package(s) when process is canceled
+          /* eslint-disable-next-line max-len */
           throw new Error(`npx canceled due to missing packages and no YES option: ${JSON.stringify(missingPackages)}`)
         }
 

--- a/workspaces/libnpmexec/lib/index.js
+++ b/workspaces/libnpmexec/lib/index.js
@@ -245,9 +245,11 @@ const exec = async (opts) => {
 
     if (add.length) {
       if (!yes) {
+        const missingPackages = add.map(a => `${a.replace(/@$/, '')}`)
         // set -n to always say no
         if (yes === false) {
-          throw new Error('canceled')
+          // Error message lists missing package(s) when process is canceled
+          throw new Error(`npx canceled due to missing packages and no YES option: ${JSON.stringify(missingPackages)}`)
         }
 
         if (noTTY() || ciInfo.isCI) {
@@ -257,8 +259,7 @@ const exec = async (opts) => {
             add.map((pkg) => pkg.replace(/@$/, '')).join(', ')
           }`)
         } else {
-          const addList = add.map(a => `  ${a.replace(/@$/, '')}`)
-            .join('\n') + '\n'
+          const addList = missingPackages.join('\n') + '\n'
           const prompt = `Need to install the following packages:\n${
           addList
         }Ok to proceed? `

